CMakeLists.txt

```diff
@@ -40,8 +40,8 @@ endif()
 # filenames)
 file(
   STRINGS src/config.inc CBMC_VERSION
-  REGEX "CBMC_VERSION = (.*)")
-string(REGEX REPLACE "CBMC_VERSION = (.*)" "\\1" CBMC_VERSION ${CBMC_VERSION})
+  REGEX "CBMC_VERSION = ([0-9.]+).*")
+string(REGEX REPLACE "CBMC_VERSION = ([0-9.]+).*" "\\1" CBMC_VERSION ${CBMC_VERSION})
```

Comment:
This section is to avoid a problem with CMake.

Essentially CMake accepts version of type `major[.minor[.patch[.tweak]]]` and fails if anything else is fed.

Before we were passing anything specified in `config.inc` as version and leaving CMake to validate it.

In this way we are just passing the part of the version that is composed of digits and `.`s, so `6.0.0-preview` is considered a valid version and for CMake it is interpreted as `6.0.0` allowing CBMC itself to use the full version specified in `config.inc`.